fcrepo-persistence-ocfl/src/main/java/org/fcrepo/persistence/ocfl/impl/FcrepoOcflObjectSessionWrapper.java

```diff
@@ -0,0 +1,176 @@
+ *
+ * @author pwinckles
+ */
+public class FcrepoOcflObjectSessionWrapper implements OcflObjectSession {
+
+    private final OcflObjectSession inner;
+
+    /**
+     * @param inner the session to wrap
+     */
+    public FcrepoOcflObjectSessionWrapper(final OcflObjectSession inner) {
+        this.inner = inner;
+    }
+
+    @Override
+    public String sessionId() {
+        return inner.sessionId();
+    }
+
+    @Override
+    public String ocflObjectId() {
+        return inner.ocflObjectId();
+    }
+
+    @Override
+    public void writeResource(final ResourceHeaders headers, final InputStream content) {
+        exec(() -> inner.writeResource(headers, content));
+    }
+
+    @Override
+    public void deleteContentFile(final ResourceHeaders headers) {
+        exec(() -> inner.deleteContentFile(headers));
+    }
+
+    @Override
+    public void deleteResource(final String resourceId) {
+        exec(() -> inner.deleteResource(resourceId));
+    }
+
+    @Override
+    public boolean containsResource(final String resourceId) {
+        return exec(() -> inner.containsResource(resourceId));
+    }
+
+    @Override
+    public ResourceHeaders readHeaders(final String resourceId) {
+        return exec(() -> inner.readHeaders(resourceId));
+    }
+
+    @Override
+    public ResourceHeaders readHeaders(final String resourceId, final String versionNumber) {
+        return exec(() -> inner.readHeaders(resourceId, versionNumber));
+    }
+
+    @Override
+    public ResourceContent readContent(final String resourceId) {
+        return exec(() -> inner.readContent(resourceId));
+    }
+
+    @Override
+    public ResourceContent readContent(final String resourceId, final String versionNumber) {
+        return exec(() -> inner.readContent(resourceId, versionNumber));
+    }
+
+    @Override
+    public List<OcflVersionInfo> listVersions(final String resourceId) {
+        return exec(() -> inner.listVersions(resourceId));
+    }
+
+    @Override
+    public Stream<ResourceHeaders> streamResourceHeaders() {
+        return exec(inner::streamResourceHeaders);
+    }
+
+    @Override
+    public void versionCreationTimestamp(final OffsetDateTime timestamp) {
+        inner.versionCreationTimestamp(timestamp);
+    }
+
+    @Override
+    public void versionAuthor(final String name, final String address) {
+        inner.versionAuthor(name, address);
+    }
+
+    @Override
+    public void versionMessage(final String message) {
+        inner.versionMessage(message);
+    }
+
+    @Override
+    public void commitType(final CommitType commitType) {
+        inner.commitType(commitType);
+    }
+
+    @Override
+    public void commit() {
+        exec(inner::commit);
+    }
+
+    @Override
+    public void abort() {
+        exec(inner::abort);
+    }
+
+    @Override
+    public boolean isOpen() {
+        return inner.isOpen();
+    }
+
+    @Override
+    public void close() throws Exception {
```

Comment:
The "throws" clause can now be removed, no?

Reply:
Not cleanly until I update `fcrepo-storage-ocfl` because the exception is still on the interface's signature.